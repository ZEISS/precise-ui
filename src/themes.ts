import * as colors from './colors';
import {
  Breakpoints,
  PreciseFullTheme,
  ButtonThemeSettings,
  FlyoutStyling,
  MetroInfoTileStyling,
  AccordionCardStyling,
  ActionButtonThemeSettings,
} from './common';
import { distance } from './distance';
import { remCalc } from './utils/remCalc';

const colorCycle = [
  colors.indigo,
  colors.green,
  colors.orangeNeon,
  colors.brightLemon,
  colors.cyan,
  colors.grey1,
  colors.orange,
  colors.azur,
  colors.purpleRed,
  colors.greenNeon,
];

export const breakpoints: Breakpoints = {
  medium: 740, // tablet
  large: 980, // desktop
  xLarge: 1200, // hd desktop
  max: 1800, // full hd desktop
};

export const buttonPrimary: ButtonThemeSettings = {
  background: colors.cyan,
  hoverBackground: colors.ocean,
  focusBackground: colors.midnight,
  disabledBackground: colors.pinkSwan,
  text: colors.white,
  hoverText: colors.white,
  focusText: colors.white,
  disabledText: colors.white,
  border: 'none',
  hoverBorder: 'none',
  focusBorder: 'none',
  disabledBorder: 'none',
  lineHeightMedium: '22px',
  lineHeightSmall: '18px',
};

export const buttonSecondary: ButtonThemeSettings = {
  background: colors.white,
  hoverBackground: colors.white,
  focusBackground: colors.white,
  disabledBackground: colors.white,
  text: colors.cyan,
  hoverText: colors.ocean,
  focusText: colors.midnight,
  disabledText: colors.pinkSwan,
  border: `1px solid ${colors.cyan}`,
  hoverBorder: `1px solid ${colors.ocean}`,
  focusBorder: `1px solid ${colors.midnight}`,
  disabledBorder: `1px solid ${colors.pinkSwan}`,
  lineHeightMedium: '20px',
  lineHeightSmall: '16px',
};

export const actionButtonWarning: ActionButtonThemeSettings = {
  iconBackground: colors.purpleRed,
  hoverIconBackground: colors.purpleRed5,
  focusIconBackground: colors.purpleRed6,
};

export const flyout: FlyoutStyling = {
  maxHeight: '600px',
  maxWidth: '300px',
  background: colors.white,
  textColor: colors.black,
  fontSize: remCalc('16px'),
};

export const metroInfoTile: MetroInfoTileStyling = {
  textColor: colors.white,
  background: colors.azur,
  size: '150px',
};

export const accordionCard: AccordionCardStyling = {
  borderColor: colors.lighterGray,
  openedBorderColor: colors.tuna,
  openedHeaderBorderColor: colors.tuna,
  openedHeaderBackground: colors.whiterSmoke,
  headerBackground: colors.white,
  headerPadding: distance.medium,
};

export const light: PreciseFullTheme = {
  flyout,
  metroInfoTile,
  accordionCard,
  ui0: colors.cyan,
  ui1: colors.white,
  ui2: colors.whiterSmoke,
  ui3: colors.whiteSmoke,
  ui4: colors.lighterGray,
  ui5: colors.tuna,
  ui6: colors.ocean,
  ui7: colors.midnight,
  ui8: colors.skyBlue,
  text0: colors.cyan,
  text1: colors.eclipse,
  text2: colors.charcoal,
  text3: colors.pinkSwan,
  text4: colors.white,
  text5: colors.darkGray,
  text6: colors.black,
  text7: colors.white,
  buttonPrimary,
  buttonSecondary,
  actionButtonWarning,
  buttonIconPosition: 'right',
  primary: colors.pacificBlue,
  secondary: colors.lighterGray,
  textDisabled: colors.pinkSwan,
  inputDisabled: colors.whiterSmoke,
  inputError: colors.purpleRed,
  colorCycle,
  headingsPadding: '0 0 0.5rem 0',
  tagBackground: colors.darkGray,
  tagColor: colors.black,
  toggleHeadBackground: colors.brightBlue,
  toggleHeadActiveBackground: colors.azur,
  breakpoints,
  accordionLine: `1px solid ${colors.lighterGray}`,
  accordionPadding: `${distance.medium}`,
  accordionContentPadding: `${distance.small} ${distance.medium} ${distance.xlarge} ${remCalc('50px')}`,
  fontFamily: 'inherit',
  tableBorder: `1px solid ${colors.lighterGray}`,
  tableLayout: 'auto',
  tableHeadPadding: `${distance.medium} ${distance.large}`,
  badgeColor: 'orange',
  badgeBackground: 'white',
  notificationColorNone: colors.tuna,
  notificationColorSuccess: colors.lightGreen,
  notificationColorInfo: colors.cyan,
  notificationColorWarning: colors.brightLemon,
  notificationColorError: colors.purpleRed,
  notificationPadding: `${distance.small} ${distance.medium}`,
  notificationBorderWidth: `1px 1px 1px 5px`,
  notificationBoxShadow: `0 4px 8px 0 rgba(0, 0, 0, 0.1)`,
  notificationTitleFontSize: remCalc('16px'),
  notificationTitleLineHeight: `22px`,
  notificationTextFontSize: remCalc('16px'),
  notificationTextLineHeight: `22px`,
  notificationIconMarginRight: distance.medium,
  highlightColor: colors.brightLemon,
};
