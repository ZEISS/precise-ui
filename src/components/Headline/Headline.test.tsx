import * as React from 'react';
import { mount, mount } from 'enzyme';
import { Headline } from './';
import { light } from '../../themes';

describe('<Headline />', () => {
  it('should render default <Headline> component', () => {
    const wrapper = mount(<Headline>h3 Headline</Headline>);
    expect(wrapper).toHaveStyleRule('font-size', '1.5rem');
    expect(wrapper).toHaveStyleRule('color', 'inherit');
  });

  it('should render h2 headline', () => {
    const wrapper = mount(<Headline level={2}>h2 Headline</Headline>);
    expect(wrapper).toMatchSnapshot();
  });

  it('with subheader should have light theme sub header dark gray color', () => {
    const wrapper = mount(<Headline subheader>h1 Headline</Headline>);
    expect(wrapper).toHaveStyleRule('color', light.text5.replace(/\s/g, ''));
  });
});
