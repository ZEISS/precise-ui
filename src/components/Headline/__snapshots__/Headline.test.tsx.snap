// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> should render h2 headline 1`] = `
.c0 {
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: 300;
  margin: 0;
  padding: 0 0 0.5rem 0;
  font-family: inherit;
  color: inherit;
}

@media (min-width:980px) {
  .c0 {
    font-size: 2rem;
  }
}

@media (max-width:979px) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:980px) {
  .c0 {
    line-height: 2.25rem;
  }
}

@media (max-width:979px) {
  .c0 {
    line-height: 1.75rem;
  }
}

<Headline
  level={2}
>
  <styled.h2
    level={2}
  >
    <h2
      className="c0"
    >
      h2 Headline
    </h2>
  </styled.h2>
</Headline>
`;
